fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuHomepageRobot.kt

```diff
@@ -43,84 +43,374 @@ class SettingsSubMenuHomepageRobot {
+        sponsoredShortcutsButton().check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+        Log.i(TAG, "verifyHomePageView: Verified that the \"Sponsored shortcuts\" option is visible")
+        Log.i(TAG, "verifyHomePageView: Trying to verify that the \"Sponsored shortcuts\" check box is checked: $sponsoredShortcutsCheckBox")
+        if (sponsoredShortcutsCheckBox) {
+            sponsoredShortcutsButton()
+                .check(
+                    matches(
+                        hasSibling(
+                            ViewMatchers.withChild(
+                                allOf(
+                                    withClassName(CoreMatchers.endsWith("CheckBox")),
+                                    isChecked(),
+                                ),
+                            ),
+                        ),
+                    ),
+                )
+        } else {
+            sponsoredShortcutsButton()
+                .check(
+                    matches(
+                        hasSibling(
+                            ViewMatchers.withChild(
+                                allOf(
+                                    withClassName(CoreMatchers.endsWith("CheckBox")),
+                                    isNotChecked(),
+                                ),
+                            ),
+                        ),
+                    ),
+                )
+        }
+        Log.i(TAG, "verifyHomePageView: Verified that the \"Sponsored shortcuts\" check box is checked: $sponsoredShortcutsCheckBox")
+        Log.i(TAG, "verifyHomePageView: Trying to verify that the \"Jump back in\" option is visible")
+        jumpBackInButton().check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+        Log.i(TAG, "verifyHomePageView: Verified that the \"Jump back in\" option is visible")
+        Log.i(TAG, "verifyHomePageView: Trying to verify that the \"Jump back in\" toggle is checked: $jumpBackInSwitchEnabled")
+        if (jumpBackInSwitchEnabled) {
+            jumpBackInButton()
+                .check(
+                    matches(
+                        TestHelper.hasCousin(
+                            Matchers.allOf(
+                                withClassName(Matchers.endsWith("Switch")),
+                                isChecked(),
+                            ),
+                        ),
+                    ),
+                )
+        } else {
+            jumpBackInButton()
+                .check(
+                    matches(
+                        TestHelper.hasCousin(
+                            Matchers.allOf(
+                                withClassName(Matchers.endsWith("Switch")),
+                                isNotChecked(),
+                            ),
+                        ),
+                    ),
+                )
+        }
+        Log.i(TAG, "verifyHomePageView: Verified that the \"Jump back in\" toggle is checked: $jumpBackInSwitchEnabled")
```

Comment:
See above. This will always be logged, no matter the condition is true or false.